specifications/secure-launch-specification.rst

```diff
@@ -0,0 +1,719 @@
+are a set of common, general requirements that all platforms should be
+able to meet. The supplemental sections will cover any idiosyncrasies for the
+various platforms and environments supported.
+
+Platform Requirements
+---------------------
+
+| **1** - General Requirements
+| **1.1** - The SLRT **MUST** begin with the magic value `0x4452544d`.
+| **1.3** - The SLRT **SHOULD** be in contiguous physical memory.
+| **1.3.1** - A preallocated, fixed size table is **OPTIONAL** through the use
+|       of the `max_size` field.
+| **1.4** - The SLRT **SHALL** be aligned to a four-byte boundary.
+|
+| **2** - UEFI Environmnents
+| **2.1** - The SLRT **SHALL** be registed in the UEFI SystemTable.
+| **2.1.2** - The GUID for registering the table **MUST BE**
+|       `877a9b2a-0385-45d1-a034-9dac9c9e565f`.
+| **2.2** - All EFI bootloaders **SHALL** record an EFI Config record with an
+|       EFI Config Entry for each measureable configuration action or
+|       information provide to the DLME kernel.
+| **2.3** - The bootloader responsible for calling ExitBootServices() **SHALL**
+|      be responsible for calling the DLE Handler.
+| **2.3.1** - The address for the SLRT **MUST** be passed via an architecture
+|        specific register.
+|
+| **3** - x86 Platforms
+| **3.1** - The SLRT **MUST** be constructed within the 4G boundary.
+| **3.1.1** - On Intel TXT platforms the location of the SLRT **SHALL** be
+|       stored in the OS2MLE structure, see Appendix B
+| **3.1.2** - On AMD platforms the location of the SLRT **SHALL** be stored in
+|       the Secure Kernel Loader (SKL) configuration table
+|
+| **4** - Arm Platforms
+| **4.1** - *Reserved*
+
+SLRT Structure
+--------------
+
+The SLRT is constructed from a header at the beginning, followed by a list of
+Tag-Length-Value (TLV) entries. Provided below is a description of the header
+and the possible entry types that may be found in the table. For general
+portability, the definitions of each entry contain a representative C
+structure.
+
+Versioning Scheme
+~~~~~~~~~~~~~~~~~
+
+The SLRT supports revisioning at two depths, at the table level and at the individual
+entry level. The table level revision is found in the SLRT header and conveys the format
+of the SLRT header as well as what SLRT entries may be appear in the table. The entry
+level revisioning is for those that are expected may need future expansion.
+
+Revision Table
+^^^^^^^^^^^^^^
+
+This table contains the list of versions for this revision of the specification.
+
++-------------------+----------+
+|                   | Revision |
++===================+==========+
+|          SLR Table|   0x01   |
++-------------------+----------+
+|  DRTM Policy Entry|   0x01   |
++-------------------+----------+
+|   EFI Config Entry|   0x01   |
++-------------------+----------+
+
+Core Entries
+~~~~~~~~~~~~
+
+The core table entries are the set of entries that are platform and
+architecture agnostic.
```

Comment:
In various places, we talk about different agnosticism. How do we define the platform? Are we talking about microarchitecture? Or we could use TCG terminology.